bitrise.yml

```diff
@@ -1484,27 +1484,66 @@ workflows:
+            TEST_PLAN="Fennec_Enterprise_PerformanceTestPlan"
+
             echo "-- build-for-testing --"
             mkdir DerivedData
             xcodebuild -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile
-            xcodebuild build-for-testing -project /Users/vagrant/git/Client.xcodeproj \
-                -scheme Fennec_Enterprise \
-                -sdk iphoneos \
-                -destination "generic/platform=iOS" \
-                -derivedDataPath "/Users/vagrant/git/DerivedData"
+            xcodebuild build-for-testing -project $PROJECT \
+                -scheme $SCHEME \
+                -sdk $SDK \
+                -destination "$DESTINATION" \
+                -derivedDataPath "$DERIVED_DATA_DIR"
+
+            echo "-- verify build artifacts --"
+
+            if [[ ! -d "$PRODUCTS_DIR" ]]; then
+                echo "Error: $PRODUCTS_DIR was not created. Check your build steps and try again."
+                exit 1
+            fi
+
+            if [[ -z $(find "$PRODUCTS_DIR" -type f -name "$TEST_PLAN*.xctestrun") ]]; then
+                echo "Error: Missing $TEST_PLAN. Check your scheme and ensure the Test Plan is added."
+                echo "The files in $PRODUCTS_DIR are:"
+                ls -l "$PRODUCTS_DIR"
+                echo "The available test plans in this environment are:"
+                awk '/<TestPlanReference/,/<\/TestPlanReference>/' \
+                    $PROJECT/xcshareddata/xcschemes/$SCHEME.xcscheme | \
+                    grep "reference" | \
+                    cut -d '"' -f 2 | \
+                    sed -n 's|container:Tests/\(.*\).xctestplan|\1|p'
+                exit 1
+            fi
+
+            if [[ ! -d "$PRODUCTS_DIR/Fennec_Enterprise-$SDK" ]]; then
+                echo "Error: Missing sdk for iPhone OS. Check your build sdk argument and try again."
+                echo "The files in $PRODUCTS_DIR are:"
+                ls -l "$PRODUCTS_DIR"
+                echo "The available sdks in this environment are:"
+                xcodebuild -showsdks | grep -E 'iphoneos|iphonesimulator'
+                exit 1
+            fi
     - script@1.1:
         title: Create .zip artifact for Firebase Testing
         inputs:
         - content: |
+            #!/usr/bin/env bash
             set -euxo pipefail
+
             echo "-- create .zip artifact --"
-            ls /Users/vagrant/git/DerivedData
-            ls /Users/vagrant/git/DerivedData/Build/Products
+            PRODUCTS_DIR="/Users/vagrant/git/DerivedData/Build/Products"
 
-            cd DerivedData/Build/Products
-            zip -r Fennec_Enterprise.zip Fennec_Enterprise-iphoneos Fennec_Enterprise_PerformanceTestPlan_iphoneos16.4-arm64.xctestrun
-            mv Fennec_Enterprise.zip /Users/vagrant/git/Fennec_Enterprise.zip
+            ( cd "$PRODUCTS_DIR" && zip -r Fennec_Enterprise.zip Fennec_Enterprise-iphoneos Fennec_Enterprise_PerformanceTestPlan*.xctestrun )
```

Comment:
@jjSDET Have you checked the extension of the files? At one point I saw `arm64-x86.xctestrun` contained in the filename. 🤔 💭

Reply:
the wildcard in this case will be the most recent sdk that bitrise's xcode will have and the target build architecture in our build settings. 

the benefit of using a wildcard here is that it doesn't matter what sdk is used or what architecture is chosen for the build when sending this to Firebase since we're building this for a generic ios device.

the wildcard portion can ultimately be anything so long as it's valid and we should be fine. the issue is that both of the variables in that wildcard can change at anytime for a variety of reasons. that's why i've included it here.

an example `xctestrun` from the most recent bitrise attempt at running the performance test was `Fennec_Enterprise_PerformanceTestPlan_iphoneos17.0-arm64.xctestrun`.

so it should always follow the form `$SCHEME_$TEST_PLAN_$SDK-$TARGET_ARCH.xctestrun`. again, the issue is more that we don't know what Bitrise's $SDK is going to be since they update it whenever they update it. As for the $TARGET_ARCH, we maybe change that at anytime for reasons I can't anticipate, which is why i was lumping everything together in one wildcard.